src/main/java/io/netty/incubator/codec/quic/Quic.java

```diff
@@ -79,6 +90,34 @@ public static Throwable unavailabilityCause() {
         return UNAVAILABILITY_CAUSE;
     }
 
+    /**
+     * Return a {@link SegmentedDatagramPacketAllocator} that is compatible with the given {@link Channel}.
+     *
+     * @param channel   the channel type to use.
+     * @return          the allocator that can be used with the channel to configure a {@link QuicCodecBuilder}.
+     */
+    public static SegmentedDatagramPacketAllocator newSegmentedAllocator(Class<? extends Channel> channel) {
+        return newSegmentedAllocator(channel, 10);
+    }
+
+    /**
+     * Return a {@link SegmentedDatagramPacketAllocator} that is compatible with the given {@link Channel}.
+     * @param channel       the channel type to use.
+     * @param maxNumPackets the maximum number of segmented packets to send in one go.
+     * @return              the allocator that can be used with the channel to configure a {@link QuicCodecBuilder}.
+     */
+    public static SegmentedDatagramPacketAllocator newSegmentedAllocator(
+            Class<? extends Channel> channel, int maxNumPackets) {
+        ObjectUtil.checkPositive(maxNumPackets, "maxNumPackets");
+        if (EPOLL_DATAGRAM_CHANNEL_CLASS != null && EPOLL_DATAGRAM_CHANNEL_CLASS.isAssignableFrom(channel) &&
+                // If EPOLL_DATAGRAM_CHANNEL_CLASS is assignable we know that the user has the right classes on the
+                // classpath.
+                EpollSegmentedDatagramPacketAllocator.isSupported()) {
+            return new EpollSegmentedDatagramPacketAllocator(maxNumPackets);
```

Comment:
@chrisvest do you have any docs on that ? I think I never implemented something like that before :)

Reply:
https://docs.oracle.com/javase/6/docs/api/java/util/ServiceLoader.html

Reply:
Basically, you have `META-INF/services/io.netty.SomeInterface` text files that contain fully qualified named of class in the given jar file, that implement that interface. Those classes need to have a public no-arg constructor. Then those classes can be loaded and objects instantiated by `ServiceLoader.load(SomeInterface.class)`.

Reply:
The loader may find zero to many implementations, and it's up to each use case how to deal with that. Since you get object instances back, you can have methods on them to help query which one is supported or compatible, or matches some configuration.